sycl/test/extensions/annotated_usm/annotated_usm.cpp

```diff
@@ -0,0 +1,243 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -fsyntax-only -Xclang -verify -Xclang -verify-ignore-unexpected=note %s
+// expected-no-diagnostics
+
+// Common tests for annotated USM allocation
+
+#include "sycl/sycl.hpp"
+
+#include <iostream>
+
+// clang-format on
+
+using namespace sycl::ext::oneapi::experimental;
+using namespace sycl::ext::intel::experimental;
+using alloc = sycl::usm::alloc;
+
+constexpr int N = 10;
+
+void TestAlloc(sycl::queue &q) {
```

Comment:
Can you organize the test cases in this file, such that it's more clear what coverage they're providing?
I feel like some of these test cases are redundant or not providing as much additional coverage as they could be. And I also think some coverage is missing.

I would like to see test cases for the following:

1. Runtime properties do not appear on the returned annotated_ptr
2. All compile-time constant properties do appear on the returned annotated_ptr
3. The usm_kind property will get automatically added to the returned annotated_ptr, including if the optional properties argument is omitted.
4. Passing a redundant usm_kind property to a function (e.g., passing `usm_kind<alloc::device`> to malloc_device_annotated) does not result in an error.
5. The aliases for the usm_kind properties, e.g., `usm_kind_device`, can be used instead of `usm_kind<alloc::device>` and the resulting annotated_ptrs have the same type no matter which one was used.

Negative test cases:
1. Error if an unsupported property is passed
2. If you pass a usm_kind property specifying a different allocation kind, e.g., `usm_kind<alloc::host>` to `malloc_device_annotated`, it results in an error.
3. If you pass a usm_kind property specifying a different allocation kind, e.g., `usm_kind<alloc::host>` to a parameterized function where the runtime argument is `alloc::device`, an exception is thrown.
4. If the properties object contains two usm_kind properties, of different allocation kinds, an error is reported.

Reply:
Most of the cases have been added, except:
Postive case 5: `usm_kind<alloc::device>` is an expression instead of a type, so the alias does not apply
negative case 1,4:  these are supposed to be `annotated_ptr` test cases

Reply:
Updates: 
- aliases for usm_kind<device>, usm_kind<host> and usm_kind<shared> have been added
- negative case 1: error out when the input property list of the malloc function contains invalid compile-time or runtime property
  - Invalid runtime property: an error will be issued on the 
  > File /tmp/wangdi4/frontend/w3/llvm/build/bin/../include/sycl/ext/oneapi/experimental/annotated_usm/alloc_util.hpp Line 104: static assertion failed due to requirement '!detail::IsRuntimeProperty<sycl::_V1::ext::intel::experimental::cache_config>::value || IsRuntimePropertyValid<sycl::_V1::ext::intel::experimental::cache_config>::value': Found invalid runtime property in the property list.
  - Invalid compile-time property: two errors will be issued, one on the malloc and one on the annotated_ptr

- negative case 4: this is invalid for any property list (which has been verified in other lit tests), including a property list used on malloc . So for the code
```
properties PL{propertyA<1>, propertyA<2>};  // Line 1
...
auto ann_ptr = malloc_annotated(... PL);   // Line 2
```
the error is issued at line 1 instead of line 2.  This case is different from negative case 1 because this is a common error, while case 1 is malloc-specific